crates/bevy_xr_utils/src/tracking_utils.rs

```diff
@@ -0,0 +1,314 @@
+        //internal
+        app.add_systems(
+            PreUpdate,
+            update_local_floor_transforms.after(update_head_transforms),
+        );
+        //external
+        app.add_systems(
+            PreUpdate,
+            update_local_floor.after(update_local_floor_transforms),
+        );
+        //bindings
+        app.add_systems(OxrSendActionBindings, suggest_action_bindings);
+        //sync actions
+        app.add_systems(
+            PreUpdate,
+            sync_actions
+                .before(OxrActionSetSyncSet)
+                .run_if(session_running),
+        );
+        //attach sets
+        app.add_systems(XrSessionCreated, attach_set);
+        //create actions
+        app.add_systems(Startup, create_actions.run_if(session_available));
+        //left grip
+        // .add_systems(
+        //     PreUpdate,
+        //     update_space_transforms
+        //         .in_set(OxrSpaceSyncSet)
+        //         .run_if(session_running),
+        // )
+        app.add_systems(PreUpdate, update_left_grip.after(update_space_transforms));
+        app.add_systems(PreUpdate, update_right_grip.after(update_space_transforms));
+    }
+}
+
+//stage
+fn update_stage(
+    mut root_query: Query<&mut Transform, (With<XrTrackingRoot>, Without<XRTrackedStage>)>,
+    mut stage_query: Query<&mut Transform, (With<XRTrackedStage>, Without<XrTrackingRoot>)>,
+) {
+    let tracking_root_transform = root_query.get_single_mut();
+    match tracking_root_transform {
+        Ok(root) => {
+            for (mut transform) in &mut stage_query {
+                *transform = root.clone();
+            }
+        }
+        Err(_) => (),
+    }
+}
+
+//view
+#[derive(Component)]
+struct HeadXRSpace(XrReferenceSpace);
+
+fn update_head_transforms(
+    session: Res<OxrSession>,
+    default_ref_space: Res<XrPrimaryReferenceSpace>,
+    pipelined: Option<Res<Pipelined>>,
+    frame_state: Res<OxrFrameState>,
+    mut query: Query<(&mut Transform, &HeadXRSpace, Option<&XrReferenceSpace>)>,
+) {
+    for (mut transform, space, ref_space) in &mut query {
+        let ref_space = ref_space.unwrap_or(&default_ref_space);
+        let time = if pipelined.is_some() {
+            openxr::Time::from_nanos(
+                frame_state.predicted_display_time.as_nanos()
+                    + frame_state.predicted_display_period.as_nanos(),
+            )
+        } else {
+            frame_state.predicted_display_time
+        };
+        let space_location = session.locate_space(&space.0, ref_space, time);
+
+        if let Ok(space_location) = space_location {
+            let flags = OxrSpaceLocationFlags(space_location.location_flags);
+            if flags.pos_valid() {
+                transform.translation = space_location.pose.position.to_vec3();
+            }
+            if flags.rot_valid() {
+                transform.rotation = space_location.pose.orientation.to_quat();
+            }
+        }
+    }
+}
```

Comment:
This is completely unnecessary right? just insert the XrSpace as a component directly, updates that transform the same

Reply:
actually the head transform is different because it is a reference space not an action space.

Reply:
doesn't matter, space is space, you can just deref to the underlying space or just access the underlying space with `.0`

Reply:
I cant make this work without this redirection, as I dont want the user facing components to be anything other than markers.

Reply:
on marker component insert, add the XrSpace?